### PR TITLE
CLI based access request workflow

### PR DESCRIPTION
**Pull request by chrnorm**
### What changed?
Adds 
```
granted exp request latest
```

and 

```
granted exp request aws
```
commands for making Access Requests to AWS roles.

https://www.loom.com/share/f49279efc7464f268b49a0d7a1acbe94

### Why?
Improves the user experience when Granted is used with a Common Fate deployment. No impact to UX if Granted is used without CF - a user won't use these subcommands.

### How did you test it?
```bash
sudo make cli

dassume
# select a role which requires an Access Request to be made

dgranted exp request latest
# this should prompt you for a reason to access the same role
```

```bash
dgranted exp request aws
# should be able to select an account ID and role and then an access request is created
```

### Potential risks
Shouldn't affect any existing functionality as these are new subcommands

### Is patch release candidate?
No - new feature

### Link to relevant docs PRs
https://github.com/common-fate/docs/pull/223

### CODE REVIEW

File: pkg/cache/migrations/20230316153630_init.down.sql
Comment (JoshuaWilkes) [inline pkg/cache/migrations/20230316153630_init.down.sql:1]: @chrnorm did you remove this dependency?
Seems like the cache migrations are unused

File: pkg/granted/exp/request/request.go [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+		&awsCommand,
+		&latestCommand,
+	},
+}
+
+var awsCommand = cli.Command{
+	Name:  "aws",
+	Usage: "Request access to an AWS role",
+	Flags: []cli.Flag{
+		&cli.StringFlag{Name: "account", Usage: "the AWS account ID"},
```
Comment (JoshuaWilkes) [inline pkg/granted/exp/request/request.go:46]:
```suggestion
		&cli.StringFlag{Name: "account", Usage: "The AWS account ID"},
```

File: pkg/granted/exp/request/request.go [hunk 2]
```diff
@@ -0,0 +1,580 @@
...
+		&cli.StringFlag{Name: "role", Usage: "the AWS role"},
```
Comment (JoshuaWilkes) [inline pkg/granted/exp/request/request.go:47]:
```suggestion
		&cli.StringFlag{Name: "role", Usage: "The AWS role"},
```

File: pkg/granted/exp/request/request.go [hunk 3]
```diff
@@ -0,0 +1,580 @@
...
+		&cli.StringFlag{Name: "reason", Usage: "a reason for access"},
```
Comment (JoshuaWilkes) [inline pkg/granted/exp/request/request.go:48]:
```suggestion
		&cli.StringFlag{Name: "reason", Usage: "A reason for access"},
```

File: pkg/granted/exp/request/request.go [hunk 4]
```diff
@@ -0,0 +1,580 @@
...
+			reason:  c.String("reason"),
+		})
+	},
+}
+
+var latestCommand = cli.Command{
+	Name:  "latest",
+	Usage: "Request access to the latest AWS role you attempted to use",
+	Flags: []cli.Flag{
+		&cli.StringFlag{Name: "reason", Usage: "a reason for access"},
```
Comment (JoshuaWilkes) [inline pkg/granted/exp/request/request.go:63]:
```suggestion
		&cli.StringFlag{Name: "reason", Usage: "A reason for access"},
```

**Review by JoshuaWilkes — CHANGES_REQUESTED** [review]

**Review by JoshuaWilkes — CHANGES_REQUESTED** [review]
I only had a comment about the sql migration files.

The rest looks fine to me

**Comment by JoshuaWilkes** [comment]
Testing this, I was able to get through requesting access ok by following the prompts.

I had to first request access and have it be denied after the web based aws flow. (I expected this, but it's something we can look at improving/ autoprompting to request)

After requesting I was shown a prompt 

image
That said I had access, however because this command runs as a subcommand of granted, it is unable to write credentials to my terminal environment.

I know I needed to then run `assume` but a regular user may get lost at this point.
We can add another info login here with `run assume latest`

File: pkg/cache/migrations/20230316153630_init.down.sql
Comment (chrnorm) [reply pkg/cache/migrations/20230316153630_init.down.sql:1]: Thanks, I've cleaned this up

**Review by JoshuaWilkes — APPROVED** [review]

**Pull request merged**